crates/tags/src/signed.rs

```diff
@@ -0,0 +1,183 @@
+        FromRequest,
+    };
+    use std::ops::Deref;
+    use std::str::FromStr;
+
+    const NAKED_ENTRY: &'static str = r#"{
+        "digest": {"sha-256": "4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo="}
+    }
+    "#;
+
+    const FLATTENED_ENTRY: &'static str = r#"{
+        "digest": {"sha-256": "4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo="},
+        "payload": "e+KAnGhhc2jigJ06InNoYS0yNTY9OjRSRWp4UTR5cnFVVmljZlNLWU5PL2NGOXpOajVBTmJ6Z0RadDMvaDNReG89OiJ9",
+        "header": { "kid": "e9bc097a-ce51-4036-9562-d2ade882db0d" },
+        "signature": "ltGzhlEG0f7cyNJVjoiVECE6RJINln5rmrLmMqyAOno="
+    }
+    "#;
+
+    const FLATTENED_PROTECTED: &'static str = r#"{
+      "digest": {"sha-256": "4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo="},
+      "payload": "e+KAnGhhc2jigJ06InNoYS0yNTY9OjRSRWp4UTR5cnFVVmljZlNLWU5PL2NGOXpOajVBTmJ6Z0RadDMvaDNReG89OiJ9",
+      "protected": "eyJjdHkiOiJhcHBsaWNhdGlvbi92bmQuZHJhd2JyaWRnZS5lbnRyeS52MStqc29uIn0K",
+      "signature": "ltGzhlEG0f7cyNJVjoiVECE6RJINln5rmrLmMqyAOno="
+    }
+    "#;
+
+    const NESTED_SIGNATURES: &'static str = r#"{
+      "digest": {"sha-256": "4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo="},
+      "payload": "e+KAnGhhc2jigJ06InNoYS0yNTY9OjRSRWp4UTR5cnFVVmljZlNLWU5PL2NGOXpOajVBTmJ6Z0RadDMvaDNReG89OiJ9",
+      "signatures": [{
+        "protected": "eyJjdHkiOiJhcHBsaWNhdGlvbi92bmQuZHJhd2JyaWRnZS5lbnRyeS52MStqc29uIn0K",
+        "signature": "ltGzhlEG0f7cyNJVjoiVECE6RJINln5rmrLmMqyAOno="
+      }]
+    }"#;
+
+    async fn test_deserialization(body: &str, mime: &str) {
+        let mut req = Request::new(
+            Method::Put,
+            Url::parse("http://example.com/_tag/TestTag").unwrap(),
+        );
+        req.set_content_type(Mime::from_str(mime).unwrap());
+        req.set_body(body);
+        let tag_value = TagValue::from_request(&mut req).await.unwrap();
+        assert_eq!(tag_value.name.deref(), "TestTag");
+        eprintln!("Kind: {:?}", tag_value.kind);
+    }
+
+    #[async_std::test]
+    async fn tag_value_naked_from_request() {
+        test_deserialization(NAKED_ENTRY, TagValue::TYPE_ENTRY).await;
+    }
+
+    #[async_std::test]
+    async fn tag_value_flattened_from_request() {
+        test_deserialization(FLATTENED_ENTRY, TagValue::TYPE_JOSE).await;
+    }
+
+    #[async_std::test]
+    async fn tag_value_protected_from_request() {
+        test_deserialization(FLATTENED_PROTECTED, TagValue::TYPE_JOSE).await;
+    }
+
+    #[async_std::test]
+    async fn tag_value_signatures_from_request() {
+        test_deserialization(NESTED_SIGNATURES, TagValue::TYPE_JOSE).await;
+    }
+
+    #[async_std::test]
+    async fn tag_name_from_request() {
+        let mut req = Request::new(
+            Method::Head,
+            Url::parse("http://example.com/_tag/TestTag").unwrap(),
+        );
+        let tag_name = TagName::from_request(&mut req).await.unwrap();
+        assert_eq!(tag_name.deref(), "TestTag");
+    }
+}
```

Comment:
Is there a reason you dropped the testing changes from https://github.com/profianinc/drawbridge/commit/69727e146c1835a6c2953aaf8f5d4a872ae9e2e1?